services/github/src/utils.js

```diff
@@ -173,6 +173,30 @@ export const tailLog = (log, maxLines = 250, maxPayloadLength = 30000) => {
     .join('\n');
 };
 
+/**
+ * Github checks API call is limited to 64kb
+ * @param {string} log
+ * @param {number} headLines
+ * @param {number} tailLines
+ * @param {number} maxPayloadLength
+ * @returns string
+ */
+export const extractLog = (log, headLines = 20, tailLines = 200, maxPayloadLength = 30000) => {
+
+  const logString = ansi2txt(log).substring(0, maxPayloadLength);
```

Comment:
I think this logic here will not work as expected. `maxPayloadLength` is for the github check (max we can attache for github check), but here it would parse the log and take first 30k chars only 
This length should be applied later and probably in some recursive/iterative way. 

We want output to look like:

```
line1
line2
..
line20

... XX lines hidden .. 

lineN-200
lineN-199
..
lineN
```

and the trick is that we don't know how long last lines are so we can be creative and check if last `tailLines` fit into `maxPayloadLength-sum(line1..line20)` and if not, take less lines and check again until we fit.

Reply:
I am thinking of binary search here. The problem I see with recursive or an iterative solution is that we need some fixed step defined for how many lines we would decrement, and worst case lets say tomorrow github increases the size of the check run output from 64kb to 200kb, so as we have room for more number of lines, we wouldn't have to change our code and it would run very efficiently. What are your thoughts?

Reply:
Yeah, I think the most simple solution here is the best. 
This function already accepts `maxPayloadLength` so it shouldn't care if github increases or decreases something, it only knows how many chars to return at most. 

Let's not over-engineer. Maybe taking first 20 lines should be fixed, and then we take last `maxPayloadLength  - length(first20Lines) - somebuffer` and then just take that much data from the end of the log, and split it with newlines. What do you think?

Reply:
modified the extractLog function as you suggested, just one thing I had a doubt on, when you say someBuffer, what should be the optimal size for it?

Reply:
Thanks! By `someBuffer` I mean extra formatting that is being used in the end - new lines and "xxx lines hidden" part. Just some random number :) `42` will do

Reply:
Added the buffer